doc/srv6/srv6_static_config_hld.md

```diff
@@ -0,0 +1,165 @@
+dscp_mode = dscp_decap_mode  ; Optional, the parameter that specifies how the node should handle DSCP bits when it performs decapsulation, default "uniform", only applicable to uDT4/uDT46/uDT6 actions
+
+For example:
+    "SRV6_MY_SID_TABLE" : {
+        "FCBB:BBBB:20::" : {
+           "action": "uN"
+        },
+        "FCBB:BBBB:20:F1::" : {
+           "action": "uDT46",
+        }
+    }
+```
+
+Since SONiC and SAI do not have full support for every SRv6 behaviors, we plan to add supported SRv6 behaviors in the system gradually.
+The current list of supported SRv6 behaviors allowed to be define in CONFIG_DB is as follows:
+
+| Alias | SRv6 Behaviors |
+| :------ | :----- |
+| uN | End with NEXT-CSID |
+| uDT46 | End.DT46 with CSID |
+
+## 3.2 Bgpcfgd changes
+
+To enable automatic programming SRv6 configurations from CONFIG_DB to FRR, we need to add a new module in bgpcfgd to watch changes in **SRV6_MY_SID_TABLE** and compile the corresponding changes in FRR's configurations.
+Following the naming convention of modules in bgpcfgd, we call this new module SRv6 Manager.
+The new SRv6 Manager are supposed to verify the validity of the configuration entries coming from the CONFIG_DB.
+If it gets an invalid configuration input, it should log the event in the syslog and not compile the configuration into FRR.
+
+## 3.3 YANG Model
+The simplified version of the YANG model is defined below.
+```
+module: sonic-srv6
+  +--rw sonic-srv6
+     +--rw SRV6_MY_SID_TABLE
+        +--rw SRV6_MY_SID_TABLE_LIST* [ip_address]
+           +--rw ip_address    inet:ipv6-address
+           +--rw block_len?    uint8
+           +--rw node_len?     uint8
+           +--rw func_len?     uint8
+           +--rw arg_len?      uint8
+           +--rw action?       enumeration
+           +--rw vrf?          -> /vrf:sonic-vrf/VRF/VRF_LIST/name
+           +--rw dscp_mode?    enumeration
+```
+Refer to [sonic-yang-models](https://github.com/sonic-net/sonic-buildimage/tree/master/src/sonic-yang-models) for the YANG model defined with standard IETF syntax.
+
+## 4 Unit Test
+
+|Test Cases | Test Result |
+| :------ | :----- |
+|add config for a SID with uN action in CONFIG_DB | verify the locator config entry is created in FRR config|
+|add config for a SID with uDT46 action in CONFIG_DB | verify the opcode config entry is created in FRR config with default VRF|
+|(Negative case) add config for a SID without action in CONFIG_DB | verify that the configuration did not get into FRR config |
+|(Negative case) add config for a SID with an unsupported action in CONFIG_DB | verify that the configuration did not get into FRR config |
+|delete config for a SID with uN action in CONFIG_DB | verify the locator config entry is deleted in FRR config|
+|delete config for a SID with uDT46 action in CONFIG_DB | verify the opcode config entry for the uDT46 action is deleted in FRR config|
+
```

Comment:
@BYGX-wcr : can we add some failure UT case also. What will happen if bgpcfgd gets exception and exited ? I assume bgp/swss/syncd docker restart and everything should get reprommed but lets add the test case to confirm.

Reply:
Do you mean when bgpcfgd restarts, the SRv6 manager should retrieve all existing MY_SID config from CONFIG_DB and program them to FRR?